Frameworks/AddressBook/ABAddressBookManagerInternal.mm

```diff
@@ -18,19 +18,52 @@
 #import "ABContactInternal.h"
 #import "UWP/WindowsApplicationModelContacts.h"
 
+@interface __ABContactOperation : NSObject
+
+@property BOOL shouldDelete;
+@property WACContact* contact;
+
+- (id)initWithContact:(WACContact*)contact shouldDelete:(BOOL)shouldDelete;
+
+@end
+
+@implementation __ABContactOperation
+
+- (id)initWithContact:(WACContact*)contact shouldDelete:(BOOL)shouldDelete {
+    if (self = [super init]) {
+        self.shouldDelete = shouldDelete;
+        self.contact = contact;
+    }
+
+    return self;
+}
+
+@end
+
 @interface _ABAddressBookManager ()
 
+- (void)_setError:(CFErrorRef*)error message:(NSString*)message;
+
 @end
 
-@implementation _ABAddressBookManager
+@implementation _ABAddressBookManager {
+    NSMutableSet<WACContact*>* _toAdd;
+    NSMutableDictionary<NSString*, __ABContactOperation*>* _operations;
+    WACContactStore* _writableStore;
+    WACContactList* _contactList;
```

Comment:
It seems like you could be leaking some of these objects. I would recommend using StrongId for the NS* objects and double check WAC* objects.